### PR TITLE
feat(refactor): improve the readability of the codebase

### LINKED ISSUE #91: Spinner on browse page
### Description

A spinner can be displayed on the "Browse" page while waiting for response from API.

Should be visible for both initial load and when going to next page.

#### Steps followed and expected result

Kill apache server and then perform operation on the page.

#### Screenshots

Current spinner is not very visual and also does not handle errors.

image

### PR DESCRIPTION
**Pull request by Shruti3004**
## Description

Refactored the codebase to increase its readability and easily understood by everyone.

## Changes

- Created the widget index.js file where created the imports of all the widgets we required in the codebase.
- Refactor the widgets imports it will help us to not impot the widget again and again.
- Fixed the license headers in all the files to maintain the symmery.
- Added the comments in all the files of services, apis, components, pages, data and many more.
- Added the comments for all the imports and state variables.
- Fixed the tokens in all the apis.
- On clicking the close button of snackbars in the `deleteUploads` and `unlinkFolder` it is breaking replaced the onClose function to setShow.
- Added the small pop up for loading while changing the pagination or entries, enhances the UX and closes #91
- Added the `jsconfig.json` for the relative paths.
- modified the `.eslintrc.json` fro resolving the imports.
- Fixed all the imports to relative paths.

### CODE REVIEW

File: src/components/Header/index.jsx [hunk 1]
```diff
@@ -27,28 +30,35 @@ import {
 } from "react-bootstrap";
 import { QuestionCircleFill, PersonCircle } from "react-bootstrap-icons";
 
-import Image from "../Widgets/Image";
-import routes from "../../constants/routes";
-import { logout, isAuth, getUserName, isAdmin } from "../../shared/authHelper";
+// Widgets
+import { Image } from "../Widgets";
```
Comment (GMishx) [inline src/components/Header/index.jsx:34]: Shouldn't the following format be used as per the style guide?

```suggestion
import Image from "../Widgets";
```

File: src/components/Upload/CommonFields/OptionalAnalysis/index.jsx [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
  with this program; if not, write to the Free Software Foundation, Inc.,
  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
 
 import React from "react";
 import PropTypes from "prop-types";
-import InputContainer from "../../../Widgets/Input";
+
+// Widgets
+import { InputContainer } from "../../../Widgets";
```
Comment (GMishx) [inline src/components/Upload/CommonFields/OptionalAnalysis/index.jsx:23]: Same here?

File: src/components/Widgets/index.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along
+ with this program; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+import { Spinner } from "react-bootstrap";
```
Comment (GMishx) [inline src/components/Widgets/index.jsx:19]: And here too.

File: src/constants/auth.js [hunk 4]
```diff
@@ -16,6 +16,11 @@
...
 
+// Length of the random string for login
 export const tokenNameLength = 40;
+
+// By default the scope for user is to write
 export const tokenScope = "write";
-export const tokenExpiryDays = 2;
+
+// Authorization token expiry days
+export const tokenExpiryDays = 1;
```
Comment (GMishx) [inline src/constants/auth.js:26]: Just a question, would `.env` be a better place for these constants? And have same default value if `.env` does not exists?

File: src/pages/Help/Overview/index.jsx [hunk 5]
```diff
@@ -17,9 +17,15 @@
 */
 
 import React from "react";
-import Image from "../../../components/Widgets/Image";
+
+// Widgets
+import { Image } from "../../../components/Widgets";
```
Comment (GMishx) [inline src/pages/Help/Overview/index.jsx:22]: And here too.

File: src/pages/Home/Features.jsx [hunk 6]
```diff
@@ -1,24 +1,27 @@
...
  with this program; if not, write to the Free Software Foundation, Inc.,
  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
 
 import React from "react";
 
-// Custom imports
-import Image from "../../components/Widgets/Image";
+// Widgets
+import { Image } from "../../components/Widgets";
```
Comment (GMishx) [inline src/pages/Home/Features.jsx:22]: And here too.

File: src/services/jobs.js [hunk 7]
```diff
@@ -18,7 +18,8 @@
 
 import getJobApi from "../api/jobs";
 
-const getJob = (jobId) => {
+// Fetching the jobs
+export const getJob = (jobId) => {
```
Comment (GMishx) [inline src/services/jobs.js:22]: There are two `export` for `getJob`

**Review by GMishx — CHANGES_REQUESTED** [review]

File: src/components/Widgets/index.jsx (hunk 3 above)
Comment (Shruti3004) [reply src/components/Widgets/index.jsx:19]: This is the import from third party library

File: src/constants/auth.js (hunk 4 above)
Comment (Shruti3004) [reply src/constants/auth.js:26]: Yes sure, we can do it

Comment (GMishx) [reply src/constants/auth.js:26]: So can you please create an issue for that and do it in a new PR?

File: src/components/Header/index.jsx (hunk 1 above)
Comment (Shruti3004) [reply src/components/Header/index.jsx]: @GMishx Fixed it!

File: src/components/Upload/CommonFields/OptionalAnalysis/index.jsx (hunk 2 above)
Comment (Shruti3004) [reply src/components/Upload/CommonFields/OptionalAnalysis/index.jsx]: Fixed

File: src/constants/auth.js (hunk 4 above)
Comment (Shruti3004) [reply src/constants/auth.js:26]: Okay sure.

File: src/pages/Help/Overview/index.jsx (hunk 5 above)
Comment (Shruti3004) [reply src/pages/Help/Overview/index.jsx]: Fixed

File: src/services/jobs.js (hunk 7 above)
Comment (Shruti3004) [reply src/services/jobs.js]: Fixed

File: src/pages/Home/Features.jsx (hunk 6 above)
Comment (Shruti3004) [reply src/pages/Home/Features.jsx]: Fixed

**Review by GMishx — APPROVED** [review]
Changes looks good. Needs test.

**Comment by Shruti3004** [comment]
@sjha2048 as you suggested, I have changed all the imports to relative paths.

**Review by sjha2048 — APPROVED** [review]

**Pull request merged**